### PR TITLE
DXQ-37812-Updated to accept both string and HTML node

### PR DESCRIPTION
**Pull request by ombhagat05**
In current enchanted tooltip, title is not accepting the html element as it was expecting only string value for title. 
Updated type of title to React.ReactNode so as developer can add html element in title tag like MUI tooltip.
Note - In MUI tooltip, title type is React.ReactNode, used same to fix the issue.

### CODE REVIEW

File: src/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ export enum TooltipPlacement {
 }
 
 export type TooltipProps = MuiTooltipProps & {
-  title?: string,
+  title?: React.ReactNode, // DXQ-37812-Updated to accept both string and HTML node
```
Comment (felix-hcl) [inline src/Tooltip/Tooltip.tsx:45]: please avoid referencing non-public ticket numbers in the opensource project.
```suggestion
  title?: React.ReactNode,
```

**Review by felix-hcl — CHANGES_REQUESTED** [review]

**Comment by felix-hcl** [comment]
@ombhagat05 additionally please also add a changelog entry. Thanks!

File: src/Tooltip/Tooltip.tsx (hunk 1 above)
Comment (ombhagat05) [reply src/Tooltip/Tooltip.tsx:45]: Got it

**Comment by ombhagat05** [comment]
> @ombhagat05 additionally please also add a changelog entry. Thanks!

@felix-hcl Not sure where to add changelog entry, please tell me.

**Review by felix-hcl — APPROVED** [review]

**Comment by ombhagat05** [comment]
Please merge this PR

File: src/Tooltip/Tooltip.stories.tsx [hunk 2]
```diff
@@ -314,6 +314,30 @@ const VisualTestTemplate: StoryFn<typeof Tooltip> = (args) => {
...
+      >
+        <Button sx={{
+          position: 'absolute',
+          left: '0',
+          marginLeft: '8px',
+        }}
+        >
+          Button
+        </Button>
+      </Tooltip>
```
Comment (rubyannmatias) [inline src/Tooltip/Tooltip.stories.tsx:340]: Hello @Omkumar Bhagat I see the slight issue now
It overlaps with one of the forced hovered on state already in the Visual Template Test (this one we use for Storyshots or Chromatic or whatever snapshot testing we have in place)

I would recommend just using the Interactive Template - you can change the title there if you want to a React.ReactNode. Thanks @ombhagat05 

See overlap - let's just avoid this altogether - please use the Interactive Example for your demo.
Screenshot 2024-10-08 at 8 55 59 PM

**Review by rubyannmatias — CHANGES_REQUESTED** [review]

File: src/Tooltip/Tooltip.stories.tsx [hunk 3]
```diff
@@ -321,16 +321,41 @@ const VisualTestTemplate: StoryFn<typeof Tooltip> = (args) => {
...
+      <Tooltip
+        {...args}
+        maxwidth={327}
+        tooltipsize="Small"
+        placement="bottom-start"
```
Comment (rubyannmatias) [inline src/Tooltip/Tooltip.stories.tsx:338]:
```suggestion
```

You can omit this now. The args will be passed from Storybook control in Line 335. @ombhagat05

**Review by rubyannmatias — CHANGES_REQUESTED** [review]

File: src/Tooltip/Tooltip.stories.tsx (hunk 3 above)
Comment (ombhagat05) [reply src/Tooltip/Tooltip.stories.tsx]: Done

**Review by rubyannmatias — APPROVED** [review]

**Review by rubyannmatias — APPROVED** [review]

**Pull request merged**